unittest/mysys/byte_order-t.c

```diff
@@ -0,0 +1,131 @@
+/* Copyright (c) 2019, MariaDB
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+/**
+  @file
+
+  Unit tests for serialization and deserialization functions
+*/
+
+#include "thr_template.c"
+
+#include "my_byteorder.h"
+#include "myisampack.h"
+
+
+void do_tests()
+{
+  MY_ALIGNED(CPU_LEVEL1_DCACHE_LINESIZE)
+  uchar buf[CPU_LEVEL1_DCACHE_LINESIZE * 2];
+
+  uchar *aligned= buf;
+  uchar *not_aligned= buf + CPU_LEVEL1_DCACHE_LINESIZE - 1;
+
+#define TEST(STORE_NAME, LOAD_NAME)                                            \
+  STORE_NAME(aligned, value);                                                  \
+  assert(LOAD_NAME(aligned) == value);                                         \
+  STORE_NAME(not_aligned, value);                                              \
+  assert(LOAD_NAME(not_aligned) == value)
```

Comment:
We should follow unittest patter, that is use `ok();` function instead of `assert();`.
I also believe we should `ok(!memcmp([not_]aligned, "<expected>"))` .

And we can make it C++ template function, which would look prettier.